modules/onMath/index.js

```diff
@@ -24,29 +26,76 @@ for(var i=0;i<mathKeysToGet.length;i++) {
   mathItems[mathKeysToGet[i]] = mathjs[mathKeysToGet[i]];
 }
 
+//
+// INPUT FULTERS
+// 
 var mathKeys = Object.keys(mathItems);
 var mathSymbols = ".,*+-/()%=";
 
 var onlySymbols = new RC("^[\\s" + REEscape(mathSymbols) + "]*$");
 var onlyNumbers = new RC(/^[\.\s\d]*$/);
-var onlyKeys = new RC("^\\s*((" + mathKeys.join(")|(") + "))+\\s*$");
-var onlyTime = new RC(/^\s*([0-9]+:?)+\s*$/);
+var onlyKeys = new RC("^((" + mathKeys.join(")|(") + "))+$");
+var onlyTime = new RC(/^(:?[0-9]+:?)+$/);
 var onlyQuote = new RC(/^".+"$/);
-var funnyFractions = new RC(/^\s*(([0-9][0]?|11)\s*\/\s*(10|5|100))\s*$/);
-var plusN = new RC(/^\s*\++\d+\s*$/);
+var funnyFractions = new RC(/^(([0-9][0]?|11)\s*\/\s*(10|5|100))$/);
+var plusN = new RC(/^\++\d+$/);
 
 var ignoreRe = onlySymbols.or(onlyNumbers).or(funnyFractions).or(onlyKeys).or(plusN)
                .or(onlyQuote).or(onlyTime);
 
+//
+// OUTPUT FILTERS
+//
+var javascript = new RC("function|{|}|return|arguments|length"); // This is by no means "good"
+
+module.exports.init = function(b) {
+  bot = b;
+}
 
 module.exports.msg = function(text, from, reply, raw) {
+  text = text.trim();
+
   if(ignoreRe.test(text)) {
     return;
   }
 
   var res = MathScopeEval(text);
 
-  if(res !== null) {
+  if(res == null) {
+    return;
+  }
+
+  res = res.toString();
+
+  var resclean = res.replace(/\s/g, '');
+  var textclean = text.replace(/\s/g, '');
+
+  // If res parrots our input, filter it
+  if( res == text || resclean == text || res == textclean 
```

Comment:
Isn't it sufficient to just check `resclean == textclean` here since that should capture all other cases?